src/vs/code/electron-main/window.ts

```diff
@@ -979,6 +979,8 @@ export class CodeWindow extends Disposable implements ICodeWindow {
 		// of the group directly prevents ugly flickering from happening
 		this.touchBarGroups.forEach((touchBarGroup, index) => {
 			const commands = groups[index];
+			// Clear the segments first to prevent icon or label remaining
+			touchBarGroup.segments = [];
```

Comment:
There seems to be a bug in electron where it doesn't correctly clear icons or labels when segments are replaced (See: https://github.com/electron/electron/issues/17322). This gets around it by clearing out the segments then replacing them. No noticeable flicker when tested on my machine.

Reply:
This issue has been fixed in Electron (https://github.com/electron/electron/pull/17336) but has yet to be released.

Reply:
This has now been released in https://github.com/electron/electron/releases/tag/v3.1.7

What is the process for updating dependencies of VSCode? Are there any guidelines that need to be followed?